### PR TITLE
Block packet size limit

### PR DESCRIPTION
**Pull request by arkpar**
Add a limit on how much block body data to send in response.

### CODE REVIEW

File: client/network/src/protocol.rs [hunk 1]
```diff
@@ -762,8 +766,9 @@ impl<B: BlockT, H: ExHashT> Protocol<B, H> {
 		let get_justification = request
 			.fields
 			.contains(message::BlockAttributes::JUSTIFICATION);
+		let mut total_size = 0;
 		while let Some(header) = self.context_data.chain.header(id).unwrap_or(None) {
-			if blocks.len() >= max {
+			if blocks.len() >= max || total_size > MAX_BODIES_BYTES {
```
[the remark this answers is not in the document]
Comment (arkpar) [reply client/network/src/protocol.rs:770]: If it hits the hard lower-level protocol packet limit after that, it still won't make it through. There should be either
* A hard limit on block body size that is shared by the network and the block imported/producer.
* Or no hard low-level packet limit.

**Comment by bkchr** [comment]
What is the status of this pr?

**Comment by arkpar** [comment]
May be reviewed

**Comment by bkchr** [comment]
If it is not used anymore, why merge it? :sweat_smile:

**Comment by arkpar** [comment]
Updated for the new block data handler. 
Also fixed a bug in the new handler, where a missing block body would result in a response with empty block body, causing an import error on the receiving side.

**Comment by tomaka** [comment]
> Also fixed a bug in the new handler, where a missing block body would result in a response with empty block body, causing an import error on the receiving side.

Your fix is obviously more correct, but I remember doing this on purpose because of one specific network test which failed if you didn't do that.
As long as the tests pass, everything is good.

**Pull request merged**